starlette/staticfiles.py

```diff
@@ -109,9 +105,30 @@ async def get_response(self, path: str, scope: Scope) -> Response:
         Returns an HTTP response, given the incoming path, method and request headers.
         """
         if scope["method"] not in ("GET", "HEAD"):
-            return PlainTextResponse("Method Not Allowed", status_code=405)
+            raise HTTPException(status_code=405)
 
-        full_path, stat_result = await self.lookup_path(path)
+        try:
+            full_path, stat_result = await anyio.to_thread.run_sync(
+                self.lookup_path, path
+            )
+        except (FileNotFoundError, NotADirectoryError):
+            if self.html:
+                # Check for '404.html' if we're in HTML mode.
+                full_path, stat_result = await anyio.to_thread.run_sync(
+                    self.lookup_path, "404.html"
+                )
+                if stat_result is not None and stat.S_ISREG(stat_result.st_mode):
```

Comment:
I also prefer "is not None" in cases like this, but just below we have the same logic in a different format.
```suggestion
                if stat_result and stat.S_ISREG(stat_result.st_mode):
```